lint-requirements.txt

```diff
@@ -0,0 +1,5 @@
+black>=24.4.2
+# Lock to v18.x until we can upgrade our code to meet v19's formatting standards.
+clang-format~=18.1
+ruff>=0.4.4
+yamllint>=1.35.1

```

Comment:
```suggestion
# Lock to v18.x until we can upgrade our code to meet v19's formatting standards.
clang-format~=18.1
clang-tidy>=19.1.0
yamllint>=1.35.1
```

* Removed black and ruff since we don't have black or ruff config in this repo (yet).
* Added clang-tidy.